str.go

```diff
@@ -0,0 +1,69 @@
+package merkletree
+
+import (
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous epochs, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	treeRoot    *interiorNode
+	epoch       int64
+	prevEpoch   int64
+	prevStrHash []byte
+	sig         []byte
+	policies    []byte
+	prev        *SignedTreeRoot
+}
+
+func (m *MerkleTree) generateSTR(ep int64, prevEp int64, prevHash []byte) *SignedTreeRoot {
```

Comment:
This probably should be a method of a `MerkleTree`, rather a constructor like `NewSTR`.